pkg/controller/argocd/role.go

```diff
@@ -36,10 +36,15 @@ func generateResourceName(argoComponentName string, cr *argoprojv1a1.ArgoCD) str
 	return cr.Name + "-" + argoComponentName
 }
 
+// generateClusterResourceName generates unique names for cluster scoped resources
+func generateClusterResourceName(argoComponentName string, cr *argoprojv1a1.ArgoCD) string {
```

Comment:
Here is where I got confused.   The resource is actually not a cluster scoped resource.  It is a namespace scoped resource, right?   Can you please fix the comment and rename the function to something like "generateUniqueResourceName".